src/main/scala/mvp2/actors/Planner.scala

```diff
@@ -15,31 +19,83 @@ class Planner(settings: Settings) extends CommonActor {
 
   var nextTurn: Period = Period(KeyBlock(), settings)
   val keyKeeper: ActorRef = context.actorOf(Props(classOf[KeyKeeper]), "keyKeeper")
-  val myKeys: KeyPair = ECDSA.createKeyPair
-  var allPublicKeys: Set[PublicKey] = Set.empty[PublicKey]
+  var myPublicKey: ByteString = ByteString.empty
+  var allPublicKeys: Set[ByteString] = Set()
   var nextPeriod: Period = Period(KeyBlock(), settings)
-  if (settings.canPublishBlocks)
-    context.system.scheduler.schedule(0 seconds, settings.plannerHeartbeat milliseconds, self, Tick)
+  var lastBlock: KeyBlock = KeyBlock()
+  var epoch: Epoch = Epoch(SortedMap())
```

Comment:
2 variables with the same meaning
  var epoch: Epoch = Epoch(SortedMap())
  var nextEpoch: Option[Epoch] = None